Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -194,14 +199,43 @@ extension Configuration {
                 }
 
             case let .default(disabled: disabledRules, optIn: optInRules):
-                if rule is OptInRule.Type, Set(optInRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is not enabled on " +
-                        "'\(Key.optInRules.rawValue)'.")
-                } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is disabled on " +
-                        "'\(Key.disabledRules.rawValue)'.")
+                validateConfiguredRulesAreEnabled(
+                    message: message,
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    rule: rule
+                )
+            }
+        }
+    }
+
+    private static func validateConfiguredRulesAreEnabled(
+        message: String,
+        parentConfiguration: Configuration?,
+        disabledRules: Set<String>,
+        optInRules: Set<String>,
+        rule: Rule.Type
+    ) {
+        if rule is OptInRule.Type {
+            var allOptInRules = optInRules
+            if let parentConfiguration {
+                switch parentConfiguration.rulesMode {
+                case .allEnabled:
+                    return
```

Comment:
Don't we have to check here that all rules minus the rules disabled in the child configuration contain the rule's ID?

Reply:
So yes, but #4859 means that opt-in rules cannot be disabled in a child configuration in some circumstances, which may change the calculation slightly.

I have a fix for that at #4863, which seems right, but I'm still not sure the current behaviour is intentional somehow.

Reply:
This should now be resolved